ruby/engine/test/RubyEngine_GTest.cpp

```diff
@@ -0,0 +1,106 @@
+  }
+
+ protected:
+  // initialize for each test
+  virtual void SetUp() override {
+    std::vector<std::string> args;
+    thisEngine = std::make_unique<openstudio::ScriptEngineInstance>("rubyengine", args);
+
+    (*thisEngine)->registerType<openstudio::measure::OSMeasure*>("openstudio::measure::OSMeasure *");
+    (*thisEngine)->registerType<openstudio::measure::ModelMeasure*>("openstudio::measure::ModelMeasure *");
+    (*thisEngine)->exec("OpenStudio::init_rest_of_openstudio()");
+  }
+  // tear down after each test
+  virtual void TearDown() override {
+    // Want to ensure the engine is reset regardless of the test outcome, or it may throw a segfault
+    thisEngine->reset();
+  }
+
+  std::unique_ptr<openstudio::ScriptEngineInstance> thisEngine;
+};
+
+TEST_F(RubyEngineFixture, BadMeasure) {
+
+  const std::string classAndDirName = "BadMeasure";
+
+  const auto scriptPath = getScriptPath(classAndDirName);
+  auto measureScriptObject = (*thisEngine)->loadMeasure(scriptPath, classAndDirName);
+  auto* measurePtr = (*thisEngine)->getAs<openstudio::measure::ModelMeasure*>(measureScriptObject);
+
+  ASSERT_EQ(measurePtr->name(), "Bad Measure");
+
+  std::string expected_exception = fmt::format(R"(SWIG director method error. RuntimeError: oops
+["/Users/julien/Software/Others/OpenStudio2/ruby/engine/test/BadMeasure/measure.rb:12:in `another_method'", "/Users/julien/Software/Others/OpenStudio2/ruby/engine/test/BadMeasure/measure.rb:16:in `arguments'"]
+
+Traceback:
+{0}:12:in `another_method'
+{0}:16:in `arguments')",
+                                               scriptPath.generic_string());
+
+  openstudio::model::Model model;
+  try {
+    measurePtr->arguments(model);
+    ASSERT_FALSE(true) << "Expected measure arguments(model) to throw";
+  } catch (std::exception& e) {
+    std::string error = e.what();
+    EXPECT_EQ(expected_exception, error);
+  }
+}
+
+TEST_F(RubyEngineFixture, WrongMethodMeasure) {
+
+  const std::string classAndDirName = "WrongMethodMeasure";
+
+  const auto scriptPath = getScriptPath(classAndDirName);
+  auto measureScriptObject = (*thisEngine)->loadMeasure(scriptPath, classAndDirName);
+  auto* measurePtr = (*thisEngine)->getAs<openstudio::measure::ModelMeasure*>(measureScriptObject);
+
+  ASSERT_EQ(measurePtr->name(), "Wrong Method Measure");
+
+  std::string expected_exception =
+    fmt::format(R"(SWIG director method error. NoMethodError: undefined method `nonExistingMethod' for #<OpenStudio::Model::Model:ADDRESS>
+["/Users/julien/Software/Others/OpenStudio2/ruby/engine/test/WrongMethodMeasure/measure.rb:12:in `arguments'"]
+
+Traceback:
+{0}:12:in `arguments')",
+                scriptPath.generic_string());
+
+  openstudio::model::Model model;
+  try {
+    measurePtr->arguments(model);
+    ASSERT_FALSE(true) << "Expected measure arguments(model) to throw";
+  } catch (std::exception& e) {
+    std::string error = e.what();
+    EXPECT_EQ(expected_exception, stripAddressFromErrorMessage(error));
```

Comment:
I gotta strip the "0x00283498743eff" address here because it naturally changes everry time